src/data/redux.test.jsx

```diff
@@ -185,59 +232,110 @@ describe('Data layer integration tests', () => {
     const readyToSubmitAttempt = Factory.build('attempt', { attempt_status: ExamStatus.READY_TO_SUBMIT });
     const readyToSubmitExam = Factory.build('exam', { attempt: readyToSubmitAttempt });
 
-    it('Should stop exam, and update attempt and exam', async () => {
-      axiosMock.onGet(fetchExamAttemptsDataUrl).replyOnce(200, { exam, active_attempt: attempt });
-      axiosMock.onGet(fetchExamAttemptsDataUrl).reply(200, { exam: readyToSubmitExam, active_attempt: {} });
-      axiosMock.onPut(updateAttemptStatusUrl).reply(200, { exam_attempt_id: readyToSubmitAttempt.attempt_id });
+    describe('with edx-proctoring as a backend (no EXAMS_BASE_URL)', () => {
+      beforeEach(() => {
+        mergeConfig({ EXAMS_BASE_URL: null });
+      });
 
-      await executeThunk(thunks.getExamAttemptsData(courseId, contentId), store.dispatch);
+      it('Should stop exam, and update attempt', async () => {
+        axiosMock.onGet(fetchExamAttemptsDataLegacyUrl).replyOnce(200, { exam, active_attempt: attempt });
+        axiosMock.onGet(fetchExamAttemptsDataLegacyUrl).reply(200, { exam: readyToSubmitExam, active_attempt: {} });
+        axiosMock.onPut(updateAttemptStatusLegacyUrl).reply(200, { exam_attempt_id: readyToSubmitAttempt.attempt_id });
+
+        await executeThunk(thunks.getExamAttemptsData(courseId, contentId), store.dispatch);
+        let state = store.getState();
+        expect(state.examState.activeAttempt.attempt_status).toBe(ExamStatus.STARTED);
+
+        await executeThunk(thunks.stopExam(), store.dispatch, store.getState);
+        state = store.getState();
+        expect(state.examState.exam.attempt.attempt_status).toBe(ExamStatus.READY_TO_SUBMIT);
+        expect(axiosMock.history.put[0].url).toEqual(updateAttemptStatusLegacyUrl);
+        expect(axiosMock.history.put[0].data).toEqual(JSON.stringify({ action: 'stop' }));
```

Comment:
added a bunch of checks like this to the original tests. In many cases our assert was just checking that the state is updated which is really only testing that we put the right mock response into the second fetchAttempts call.